py4syn/epics/DxpClass.py

```diff
@@ -104,25 +113,42 @@ def getValueChannel(self, **kwargs):
         channel = kwargs['channel']
         c = int(channel[3]) - 1
         if(len(channel) > 4):
-            r = int(channel[5])
-            return self.pvDxpRois[c][r]
+            r = int(channel[6])
+            return self.pvDxpRois[c][r].get()
         else:
             self.saveSpectrum(c, **kwargs)
             return 1.0
 
     # save the spectrum intensity in a mca file
+    # or an hdf file
     def saveSpectrum(self, ch, **kwargs):
-        fileName = self.fileName
-        idx = 0
-        if(fileName):
-            spectrum = self.pvDxpChannels[ch].get(as_numpy=True)
-            prefix = fileName.split('.')[0]
-            while os.path.exists('%s_%s%d_%04d.mca' % (prefix, self.dxpType,
-                                                       ch, idx)):
-                idx += 1
-            fileName = '%s_%s%d_%04d.mca' % \
-                       (prefix, self.dxpType, ch, idx)
-        np.savetxt(fileName, spectrum, fmt='%f')
+        self.ch = ch
```

Comment:
This method is a bit confusing, would you mind to show a code snippet with a use case for it?